simplifier/operations/bitwise.py

```diff
@@ -446,11 +446,30 @@ def _simple_folding(self):
         maximum = Constant.create_maximum_value(self.world, max([operand.size for operand in self.operands]))
         if self._has_collision() or maximum in self.constants:
             self.replace_operands([maximum])
+        if self.has_collisions_of_another_kind():
+            one = Constant(self.world, 1, max([operand.size for operand in self.operands]))
+            self.replace_operands([one])
         for constant in self.constants:
             # remove any zero constant, unless its the last operand
             if constant.unsigned == 0 and len(self.operands) > 1:
                 self.remove_operand(constant)
 
+    def has_collisions_of_another_kind(self):
+        """
+        Check for Equal/Not-Equal Collisions.
+
+        (a == 0) | (a != 0) = 1
+        """
+        if len(self.operands) == 2:
+            from .boolean import Equal, Unequal  # avoid circular import
+
+            if isinstance(self.operands[0], (Equal, Unequal)):
+                neg = self.world.bool_negate(self.operands[0])
+                neg = neg.simplify()
+                if self.world.compare(neg, self.operands[1]):
+                    return True
+        return False
```

Comment:
I have tried commenting out the use of this function, and it seems that the pytests for ```(a == c) | (a != c) = 1``` and its variants fail. Perhaps I should change the name of this function instead?

Reply:
The range simplifier is currently not part of `simplify`. I was not sure where to add it. If you call `RangeSimplifier.simplify((a == c) | (a != c))` the result should be `1`. Perhaps we should add it there, I am not sure.

Reply:
I actually tried using the RangeSimplifier for those failing tests (following how it was done for the Range Simplifier tests) and they still failed :(

Reply:
Yes, it fails because if you call in on the operation `!` then it only tries to simplify this one. 
I tried something out, that works when we rewrite 4 tests.
First:
```python
class RangeSimplifier:
.....
        if isinstance(self._world.get_definition(self._defining_variable), Operation):
            self.get_operation.simplify()
        return self.can_be_simplified
```
We should probably rename the functions here a bit, i.e., `can_be_simplified` to `can_be_ranged_simplified` and a function for the first instance with the name `can_be_simplified`
Second in `test_bool.py`: 
```python
def run_test(test: str, result: str) -> bool:
    """Run the given test case."""
    w = World()
    a, b = w.variable("a", 32), w.variable("b", 32)
    test_operation = w.from_string(test)
    w.define(a, test_operation)
    w.define(b, w.from_string(result))
    a.simplify()  # type: ignore #only change this line for the test where it is needed. Not for all.
    return w.compare(a, b)
```
Third:
```
class BaseVariable(BitVector, ABC):
    def simplify(self):
        """Simplify the term defined by the variable."""
        from simplifier.operations import BitwiseNegate
        from simplifier.range_simplifier import RangeSimplifier

        for node in list(self.world.iter_postorder(self)):
            if isinstance(node, BitwiseNegate):
                node.dissolve_negation()
        for node in list(self.world.iter_postorder(self)):
            if isinstance(node, Operation):
                RangeSimplifier.simplify(node)
```
After this, you should be able to remove this part. But I am not sure whether this is what we want. It is only a suggestions

Reply:
Nevermind, I managed to fix it to use the RangeSimplifier, and have removed ``has_collisions_of_another_kind```